### PR TITLE
Return instances of Views rather than dicts

### PR DESCRIPTION
**Pull request by fbertsch**
Also created a few more functions and cleaned up `namespaces.py`, and added some more tests for those functions.

This also removes the `_daily` tables, as right now we only create explores for ping tables and _last_seen tables (as the growth_accounting explore).

(No implementation yet of that explore, this just updates the namespaces.yaml)

### CODE REVIEW

File: generator/views.py [hunk 1]
```diff
@@ -2,57 +2,118 @@
...
-                    table["channel"] = app["app_channel"]
-                if len(references) == 1 and references[0][-2] == f"{dataset_id}_stable":
-                    # view references a single table in the stable dataset
-                    table["is_ping_table"] = True
-                elif not is_release:
-                    continue  # ignore non-ping tables from non-release datasets
+
+                table: Dict[str, str] = {"table": f"mozdata.{dataset}.{view_id}"}
+
+                if channel.get("channel"):
```
Comment (relud) [inline generator/views.py:84]: this should use the more idiomatic `in` syntax, with the added benefit that it correctly handles falsey values
```suggestion
                if "channel" in channel:
```

File: tests/test_namespaces.py [hunk 2]
```diff
@@ -39,32 +47,43 @@ def custom_namespaces(tmp_path):
...
+
         for dataset in ("glean_app", "glean_app_beta"):
-            content = dedent(
-                f"""
+            content = f"""
                 references:
                   view.sql:
                   - moz-fx-data-shared-prod.{dataset}_derived.baseline_clients_daily_v1
                 """
-            ).lstrip()
```
Comment (relud) [inline tests/test_namespaces.py:49]: why is `dedent` removed here?

Comment (fbertsch) [reply tests/test_namespaces.py:49]: I moved it to the function: https://github.com/mozilla/lookml-generator/pull/29/files#diff-771694b87a3d9b5cede55055ad91f9c22fc4fac470dd326fc6112ae662cc78acR52

File: generator/lookml.py [hunk 3]
```diff
@@ -38,6 +38,13 @@
 }
 
 
+class ViewDict(TypedDict):
```
Comment (fbertsch) [inline generator/lookml.py:41]: This is just for typing purposes. Mypy was complaining.

File: generator/lookml.py [hunk 4]
```diff
@@ -184,6 +192,9 @@ def _generate_explores(
     client, out_dir: Path, namespace: str, explores: dict
 ) -> Iterable[Path]:
     for explore_name, defn in explores.items():
+        if defn["type"] != "ping_explore":
```
Comment (fbertsch) [inline generator/lookml.py:195]: For now, only generate ping explores.

**Comment by fbertsch** [comment]
@relud apologies for the late changes. I'm working on a follow-up that updates lookml generation for all of these changes (using classes, generating different types of classes, etc.) and it made more sense for a few of those changes to live here rather than in the follow-up.

**Comment by fbertsch** [comment]
<details>
<summary>New namespaces.yaml</summary>

```
bergamot:
  canonical_app_name: Bergamot Translator
  explores:
    custom:
      type: ping_explore
      views:
        base_view: custom
  views:
    custom:
      tables:
      - table: mozdata.org_mozilla_bergamot.custom
      type: ping_view
burnham:
  canonical_app_name: Burnham
  explores:
    baseline:
      type: ping_explore
      views:
        base_view: baseline
    discovery:
      type: ping_explore
      views:
        base_view: discovery
    events:
      type: ping_explore
      views:
        base_view: events
    growth_accounting:
      type: growth_accounting_explore
      views:
        base_view: growth_accounting
    metrics:
      type: ping_explore
      views:
        base_view: metrics
    space_ship_ready:
      type: ping_explore
      views:
        base_view: space_ship_ready
    starbase46:
      type: ping_explore
      views:
        base_view: starbase46
  views:
    baseline:
      tables:
      - table: mozdata.burnham.baseline
      type: ping_view
    discovery:
      tables:
      - table: mozdata.burnham.discovery
      type: ping_view
    events:
      tables:
      - table: mozdata.burnham.events
      type: ping_view
    growth_accounting:
      tables:
      - table: mozdata.burnham.baseline_clients_last_seen
      type: growth_accounting_view
    metrics:
      tables:
      - table: mozdata.burnham.metrics
      type: ping_view
    space_ship_ready:
      tables:
      - table: mozdata.burnham.space_ship_ready
      type: ping_view
    starbase46:
      tables:
      - table: mozdata.burnham.starbase46
      type: ping_view
fenix:
  canonical_app_name: Firefox for Android
  explores:
    activation:
      type: ping_explore
      views:
        base_view: activation
    baseline:
      type: ping_explore
      views:
        base_view: baseline
    bookmarks_sync:
      type: ping_explore
      views:
        base_view: bookmarks_sync
    events:
      type: ping_explore
      views:
        base_view: events
    first_session:
      type: ping_explore
      views:
        base_view: first_session
    growth_accounting:
      type: growth_accounting_explore
      views:
        base_view: growth_accounting
    history_sync:
      type: ping_explore
      views:
        base_view: history_sync
    installation:
      type: ping_explore
      views:
        base_view: installation
    logins_sync:
      type: ping_explore
      views:
        base_view: logins_sync
    metrics:
      type: ping_explore
      views:
        base_view: metrics
    migration:
      type: ping_explore
      views:
        base_view: migration
    startup_timeline:
      type: ping_explore
      views:
        base_view: startup_timeline
    sync:
      type: ping_explore
      views:
        base_view: sync
  views:
    activation:
      tables:
      - channel: release
        table: mozdata.org_mozilla_firefox.activation
      - channel: beta
        table: mozdata.org_mozilla_firefox_beta.activation
      - channel: nightly
        table: mozdata.org_mozilla_fenix.activation
      type: ping_view
    baseline:
      tables:
      - channel: release
        table: mozdata.org_mozilla_firefox.baseline
      - channel: beta
        table: mozdata.org_mozilla_firefox_beta.baseline
      - channel: nightly
        table: mozdata.org_mozilla_fenix.baseline
      type: ping_view
    bookmarks_sync:
      tables:
      - channel: release
        table: mozdata.org_mozilla_firefox.bookmarks_sync
      - channel: beta
        table: mozdata.org_mozilla_firefox_beta.bookmarks_sync
      - channel: nightly
        table: mozdata.org_mozilla_fenix.bookmarks_sync
      type: ping_view
    events:
      tables:
      - channel: release
        table: mozdata.org_mozilla_firefox.events
      - channel: beta
        table: mozdata.org_mozilla_firefox_beta.events
      - channel: nightly
        table: mozdata.org_mozilla_fenix.events
      type: ping_view
    first_session:
      tables:
      - channel: release
        table: mozdata.org_mozilla_firefox.first_session
      - channel: beta
        table: mozdata.org_mozilla_firefox_beta.first_session
      - channel: nightly
        table: mozdata.org_mozilla_fenix.first_session
      type: ping_view
    growth_accounting:
      tables:
      - table: mozdata.org_mozilla_firefox.baseline_clients_last_seen
      type: growth_accounting_view
    history_sync:
      tables:
      - channel: release
        table: mozdata.org_mozilla_firefox.history_sync
      - channel: beta
        table: mozdata.org_mozilla_firefox_beta.history_sync
      - channel: nightly
        table: mozdata.org_mozilla_fenix.history_sync
      type: ping_view
    installation:
      tables:
      - channel: release
        table: mozdata.org_mozilla_firefox.installation
      - channel: beta
        table: mozdata.org_mozilla_firefox_beta.installation
      - channel: nightly
        table: mozdata.org_mozilla_fenix.installation
      type: ping_view
    logins_sync:
      tables:
      - channel: release
        table: mozdata.org_mozilla_firefox.logins_sync
      - channel: beta
        table: mozdata.org_mozilla_firefox_beta.logins_sync
      - channel: nightly
        table: mozdata.org_mozilla_fenix.logins_sync
      type: ping_view
    metrics:
      tables:
      - channel: release
        table: mozdata.org_mozilla_firefox.metrics
      - channel: beta
        table: mozdata.org_mozilla_firefox_beta.metrics
      - channel: nightly
        table: mozdata.org_mozilla_fenix.metrics
      type: ping_view
    migration:
      tables:
      - channel: beta
        table: mozdata.org_mozilla_firefox_beta.migration
      - channel: nightly
        table: mozdata.org_mozilla_fenix.migration
      type: ping_view
    startup_timeline:
      tables:
      - channel: release
        table: mozdata.org_mozilla_firefox.startup_timeline
      - channel: beta
        table: mozdata.org_mozilla_firefox_beta.startup_timeline
      - channel: nightly
        table: mozdata.org_mozilla_fenix.startup_timeline
      type: ping_view
    sync:
      tables:
      - channel: release
        table: mozdata.org_mozilla_firefox.sync
      - channel: beta
        table: mozdata.org_mozilla_firefox_beta.sync
      - channel: nightly
        table: mozdata.org_mozilla_fenix.sync
      type: ping_view
firefox_desktop:
  canonical_app_name: Firefox for Desktop
  explores:
    baseline:
      type: ping_explore
      views:
        base_view: baseline
    events:
      type: ping_explore
      views:
        base_view: events
    fog_validation:
      type: ping_explore
      views:
        base_view: fog_validation
    growth_accounting:
      type: growth_accounting_explore
      views:
        base_view: growth_accounting
    metrics:
      type: ping_explore
      views:
        base_view: metrics
  views:
    baseline:
      tables:
      - table: mozdata.firefox_desktop.baseline
      type: ping_view
    events:
      tables:
      - table: mozdata.firefox_desktop.events
      type: ping_view
    fog_validation:
      tables:
      - table: mozdata.firefox_desktop.fog_validation
      type: ping_view
    growth_accounting:
      tables:
      - table: mozdata.firefox_desktop.baseline_clients_last_seen
      type: growth_accounting_view
    metrics:
      tables:
      - table: mozdata.firefox_desktop.metrics
      type: ping_view
firefox_echo_show:
  canonical_app_name: Firefox for Echo Show
  explores:
    baseline:
      type: ping_explore
      views:
        base_view: baseline
    events:
      type: ping_explore
      views:
        base_view: events
    growth_accounting:
      type: growth_accounting_explore
      views:
        base_view: growth_accounting
    metrics:
      type: ping_explore
      views:
        base_view: metrics
  views:
    baseline:
      tables:
      - table: mozdata.org_mozilla_connect_firefox.baseline
      type: ping_view
    events:
      tables:
      - table: mozdata.org_mozilla_connect_firefox.events
      type: ping_view
    growth_accounting:
      tables:
      - table: mozdata.org_mozilla_connect_firefox.baseline_clients_last_seen
      type: growth_accounting_view
    metrics:
      tables:
      - table: mozdata.org_mozilla_connect_firefox.metrics
      type: ping_view
firefox_fire_tv:
  canonical_app_name: Firefox for Fire TV
  explores:
    baseline:
      type: ping_explore
      views:
        base_view: baseline
    events:
      type: ping_explore
      views:
        base_view: events
    growth_accounting:
      type: growth_accounting_explore
      views:
        base_view: growth_accounting
    metrics:
      type: ping_explore
      views:
        base_view: metrics
  views:
    baseline:
      tables:
      - table: mozdata.org_mozilla_tv_firefox.baseline
      type: ping_view
    events:
      tables:
      - table: mozdata.org_mozilla_tv_firefox.events
      type: ping_view
    growth_accounting:
      tables:
      - table: mozdata.org_mozilla_tv_firefox.baseline_clients_last_seen
      type: growth_accounting_view
    metrics:
      tables:
      - table: mozdata.org_mozilla_tv_firefox.metrics
      type: ping_view
firefox_ios:
  canonical_app_name: Firefox for iOS
  explores:
    baseline:
      type: ping_explore
      views:
        base_view: baseline
    events:
      type: ping_explore
      views:
        base_view: events
    growth_accounting:
      type: growth_accounting_explore
      views:
        base_view: growth_accounting
    metrics:
      type: ping_explore
      views:
        base_view: metrics
  views:
    baseline:
      tables:
      - channel: release
        table: mozdata.org_mozilla_ios_firefox.baseline
      - channel: beta
        table: mozdata.org_mozilla_ios_firefoxbeta.baseline
      - channel: nightly
        table: mozdata.org_mozilla_ios_fennec.baseline
      type: ping_view
    events:
      tables:
      - channel: release
        table: mozdata.org_mozilla_ios_firefox.events
      - channel: beta
        table: mozdata.org_mozilla_ios_firefoxbeta.events
      - channel: nightly
        table: mozdata.org_mozilla_ios_fennec.events
      type: ping_view
    growth_accounting:
      tables:
      - table: mozdata.org_mozilla_ios_firefox.baseline_clients_last_seen
      type: growth_accounting_view
    metrics:
      tables:
      - channel: release
        table: mozdata.org_mozilla_ios_firefox.metrics
      - channel: beta
        table: mozdata.org_mozilla_ios_firefoxbeta.metrics
      - channel: nightly
        table: mozdata.org_mozilla_ios_fennec.metrics
      type: ping_view
firefox_reality:
  canonical_app_name: Firefox Reality
  explores:
    baseline:
      type: ping_explore
      views:
        base_view: baseline
    bookmarks_sync:
      type: ping_explore
      views:
        base_view: bookmarks_sync
    events:
      type: ping_explore
      views:
        base_view: events
    growth_accounting:
      type: growth_accounting_explore
      views:
        base_view: growth_accounting
    history_sync:
      type: ping_explore
      views:
        base_view: history_sync
    logins_sync:
      type: ping_explore
      views:
        base_view: logins_sync
    metrics:
      type: ping_explore
      views:
        base_view: metrics
    session_end:
      type: ping_explore
      views:
        base_view: session_end
    sync:
      type: ping_explore
      views:
        base_view: sync
  views:
    baseline:
      tables:
      - table: mozdata.org_mozilla_vrbrowser.baseline
      type: ping_view
    bookmarks_sync:
      tables:
      - table: mozdata.org_mozilla_vrbrowser.bookmarks_sync
      type: ping_view
    events:
      tables:
      - table: mozdata.org_mozilla_vrbrowser.events
      type: ping_view
    growth_accounting:
      tables:
      - table: mozdata.org_mozilla_vrbrowser.baseline_clients_last_seen
      type: growth_accounting_view
    history_sync:
      tables:
      - table: mozdata.org_mozilla_vrbrowser.history_sync
      type: ping_view
    logins_sync:
      tables:
      - table: mozdata.org_mozilla_vrbrowser.logins_sync
      type: ping_view
    metrics:
      tables:
      - table: mozdata.org_mozilla_vrbrowser.metrics
      type: ping_view
    session_end:
      tables:
      - table: mozdata.org_mozilla_vrbrowser.session_end
      type: ping_view
    sync:
      tables:
      - table: mozdata.org_mozilla_vrbrowser.sync
      type: ping_view
firefox_reality_pc:
  canonical_app_name: Firefox Reality for PC-connected VR platforms
  explores:
    baseline:
      type: ping_explore
      views:
        base_view: baseline
    events:
      type: ping_explore
      views:
        base_view: events
    growth_accounting:
      type: growth_accounting_explore
      views:
        base_view: growth_accounting
    launch:
      type: ping_explore
      views:
        base_view: launch
    metrics:
      type: ping_explore
      views:
        base_view: metrics
  views:
    baseline:
      tables:
      - table: mozdata.org_mozilla_firefoxreality.baseline
      type: ping_view
    events:
      tables:
      - table: mozdata.org_mozilla_firefoxreality.events
      type: ping_view
    growth_accounting:
      tables:
      - table: mozdata.org_mozilla_firefoxreality.baseline_clients_last_seen
      type: growth_accounting_view
    launch:
      tables:
      - table: mozdata.org_mozilla_firefoxreality.launch
      type: ping_view
    metrics:
      tables:
      - table: mozdata.org_mozilla_firefoxreality.metrics
      type: ping_view
focus_ios:
  canonical_app_name: Firefox Focus for iOS
  explores:
    baseline:
      type: ping_explore
      views:
        base_view: baseline
    events:
      type: ping_explore
      views:
        base_view: events
    growth_accounting:
      type: growth_accounting_explore
      views:
        base_view: growth_accounting
    metrics:
      type: ping_explore
      views:
        base_view: metrics
  views:
    baseline:
      tables:
      - table: mozdata.org_mozilla_ios_focus.baseline
      type: ping_view
    events:
      tables:
      - table: mozdata.org_mozilla_ios_focus.events
      type: ping_view
    growth_accounting:
      tables:
      - table: mozdata.org_mozilla_ios_focus.baseline_clients_last_seen
      type: growth_accounting_view
    metrics:
      tables:
      - table: mozdata.org_mozilla_ios_focus.metrics
      type: ping_view
klar_ios:
  canonical_app_name: Firefox Klar for iOS
  explores:
    baseline:
      type: ping_explore
      views:
        base_view: baseline
    events:
      type: ping_explore
      views:
        base_view: events
    growth_accounting:
      type: growth_accounting_explore
      views:
        base_view: growth_accounting
    metrics:
      type: ping_explore
      views:
        base_view: metrics
  views:
    baseline:
      tables:
      - table: mozdata.org_mozilla_ios_klar.baseline
      type: ping_view
    events:
      tables:
      - table: mozdata.org_mozilla_ios_klar.events
      type: ping_view
    growth_accounting:
      tables:
      - table: mozdata.org_mozilla_ios_klar.baseline_clients_last_seen
      type: growth_accounting_view
    metrics:
      tables:
      - table: mozdata.org_mozilla_ios_klar.metrics
      type: ping_view
lockwise_android:
  canonical_app_name: Lockwise for Android
  explores:
    baseline:
      type: ping_explore
      views:
        base_view: baseline
    bookmarks_sync:
      type: ping_explore
      views:
        base_view: bookmarks_sync
    events:
      type: ping_explore
      views:
        base_view: events
    growth_accounting:
      type: growth_accounting_explore
      views:
        base_view: growth_accounting
    history_sync:
      type: ping_explore
      views:
        base_view: history_sync
    logins_sync:
      type: ping_explore
      views:
        base_view: logins_sync
    metrics:
      type: ping_explore
      views:
        base_view: metrics
    sync:
      type: ping_explore
      views:
        base_view: sync
  views:
    baseline:
      tables:
      - table: mozdata.mozilla_lockbox.baseline
      type: ping_view
    bookmarks_sync:
      tables:
      - table: mozdata.mozilla_lockbox.bookmarks_sync
      type: ping_view
    events:
      tables:
      - table: mozdata.mozilla_lockbox.events
      type: ping_view
    growth_accounting:
      tables:
      - table: mozdata.mozilla_lockbox.baseline_clients_last_seen
      type: growth_accounting_view
    history_sync:
      tables:
      - table: mozdata.mozilla_lockbox.history_sync
      type: ping_view
    logins_sync:
      tables:
      - table: mozdata.mozilla_lockbox.logins_sync
      type: ping_view
    metrics:
      tables:
      - table: mozdata.mozilla_lockbox.metrics
      type: ping_view
    sync:
      tables:
      - table: mozdata.mozilla_lockbox.sync
      type: ping_view
lockwise_ios:
  canonical_app_name: Lockwise for iOS
  explores:
    baseline:
      type: ping_explore
      views:
        base_view: baseline
    events:
      type: ping_explore
      views:
        base_view: events
    growth_accounting:
      type: growth_accounting_explore
      views:
        base_view: growth_accounting
    metrics:
      type: ping_explore
      views:
        base_view: metrics
  views:
    baseline:
      tables:
      - table: mozdata.org_mozilla_ios_lockbox.baseline
      type: ping_view
    events:
      tables:
      - table: mozdata.org_mozilla_ios_lockbox.events
      type: ping_view
    growth_accounting:
      tables:
      - table: mozdata.org_mozilla_ios_lockbox.baseline_clients_last_seen
      type: growth_accounting_view
    metrics:
      tables:
      - table: mozdata.org_mozilla_ios_lockbox.metrics
      type: ping_view
mach:
  canonical_app_name: mach
  explores:
    baseline:
      type: ping_explore
      views:
        base_view: baseline
    events:
      type: ping_explore
      views:
        base_view: events
    growth_accounting:
      type: growth_accounting_explore
      views:
        base_view: growth_accounting
    metrics:
      type: ping_explore
      views:
        base_view: metrics
    usage:
      type: ping_explore
      views:
        base_view: usage
  views:
    baseline:
      tables:
      - table: mozdata.mozilla_mach.baseline
      type: ping_view
    events:
      tables:
      - table: mozdata.mozilla_mach.events
      type: ping_view
    growth_accounting:
      tables:
      - table: mozdata.mozilla_mach.baseline_clients_last_seen
      type: growth_accounting_view
    metrics:
      tables:
      - table: mozdata.mozilla_mach.metrics
      type: ping_view
    usage:
      tables:
      - table: mozdata.mozilla_mach.usage
      type: ping_view
mozphab:
  canonical_app_name: mozphab
  explores:
    baseline:
      type: ping_explore
      views:
        base_view: baseline
    events:
      type: ping_explore
      views:
        base_view: events
    growth_accounting:
      type: growth_accounting_explore
      views:
        base_view: growth_accounting
    metrics:
      type: ping_explore
      views:
        base_view: metrics
    usage:
      type: ping_explore
      views:
        base_view: usage
  views:
    baseline:
      tables:
      - table: mozdata.mozphab.baseline
      type: ping_view
    events:
      tables:
      - table: mozdata.mozphab.events
      type: ping_view
    growth_accounting:
      tables:
      - table: mozdata.mozphab.baseline_clients_last_seen
      type: growth_accounting_view
    metrics:
      tables:
      - table: mozdata.mozphab.metrics
      type: ping_view
    usage:
      tables:
      - table: mozdata.mozphab.usage
      type: ping_view
mozregression:
  canonical_app_name: mozregression
  explores:
    baseline:
      type: ping_explore
      views:
        base_view: baseline
    events:
      type: ping_explore
      views:
        base_view: events
    growth_accounting:
      type: growth_accounting_explore
      views:
        base_view: growth_accounting
    metrics:
      type: ping_explore
      views:
        base_view: metrics
    usage:
      type: ping_explore
      views:
        base_view: usage
  views:
    baseline:
      tables:
      - table: mozdata.org_mozilla_mozregression.baseline
      type: ping_view
    events:
      tables:
      - table: mozdata.org_mozilla_mozregression.events
      type: ping_view
    growth_accounting:
      tables:
      - table: mozdata.org_mozilla_mozregression.baseline_clients_last_seen
      type: growth_accounting_view
    metrics:
      tables:
      - table: mozdata.org_mozilla_mozregression.metrics
      type: ping_view
    usage:
      tables:
      - table: mozdata.org_mozilla_mozregression.usage
      type: ping_view
reference_browser:
  canonical_app_name: Reference Browser
  explores:
    baseline:
      type: ping_explore
      views:
        base_view: baseline
    events:
      type: ping_explore
      views:
        base_view: events
    growth_accounting:
      type: growth_accounting_explore
      views:
        base_view: growth_accounting
    metrics:
      type: ping_explore
      views:
        base_view: metrics
  views:
    baseline:
      tables:
      - table: mozdata.org_mozilla_reference_browser.baseline
      type: ping_view
    events:
      tables:
      - table: mozdata.org_mozilla_reference_browser.events
      type: ping_view
    growth_accounting:
      tables:
      - table: mozdata.org_mozilla_reference_browser.baseline_clients_last_seen
      type: growth_accounting_view
    metrics:
      tables:
      - table: mozdata.org_mozilla_reference_browser.metrics
      type: ping_view
```

</details>

**Review by relud — APPROVED** [review]
r+wc

**Pull request merged**